### PR TITLE
add oonireport client

### PR DESCRIPTION
**Pull request by kelmenhorst**
The oonireport client (re-)uploads a measurement report file. This can be helpful when the measurement was not uploaded at runtime.

Usage: `./oonireport upload <file>`, where `<file>` is a json(l) file containing one OONI measurement per line.

This pull request refers to https://github.com/ooni/probe/issues/2003 and https://github.com/ooni/probe/issues/950.

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/ooni/probe-cli/blob/master/CONTRIBUTING.md)
- [X] reference issue for this pull request: 
- [ ] if you changed anything related how experiments work and you need to reflect these changes in the ooni/spec repository, please link to the related ooni/spec pull request:

### CODE REVIEW

**Comment by kelmenhorst** [comment]
- [X] added unit tests

File: internal/cmd/oonireport/oonireport.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
+package main
```
Comment (bassosimone) [inline internal/cmd/oonireport/oonireport.go:1]:
```suggestion
// Command oonireport uploads reports stored on disk to the OONI collector.
package main
```

File: internal/cmd/oonireport/oonireport.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+// toMeasurement loads an input string as model.Measurement
+func toMeasurement(s string) *model.Measurement {
+	var mm model.Measurement
+	err := json.Unmarshal([]byte(s), &mm)
+	runtimex.PanicOnError(err, "json.Unmarshal error")
+	return &mm
+}
+
+func submitAll(ctx context.Context, lines []string, subm *probeservices.Submitter) (int, error) {
```
Comment (bassosimone) [inline internal/cmd/oonireport/oonireport.go:111]:
```suggestion
// submitAll submits the measurements in input. Returns the count of submitted measurements, both
// on success and on error, and the error that occurred (nil on success).
func submitAll(ctx context.Context, lines []string, subm *probeservices.Submitter) (int, error) {
```

File: internal/cmd/oonireport/oonireport.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	ctx := context.Background()
+	sess := newSession(ctx)
+	defer sess.Close()
+
+	submitter := newSubmitter(sess, ctx)
+
+	n, err := submitAll(ctx, lines, submitter)
+	runtimex.PanicOnError(err, "error occurred while submitting")
+
+	fmt.Println("Submitted measurements: ", n)
```
Comment (bassosimone) [inline internal/cmd/oonireport/oonireport.go:144]:
```suggestion
	n, err := submitAll(ctx, lines, submitter)
	fmt.Println("Submitted measurements: ", n)
	runtimex.PanicOnError(err, "error occurred while submitting")
```

File: internal/cmd/oonireport/oonireport_test.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+func TestSubmitAllFails(t *testing.T) {
+	ctx := context.Background()
+	sess := newSession(ctx)
+	subm := newSubmitter(sess, ctx)
+	lines := readLines("testdata/testmeasurement.json")
+
+	ctx, cancel := context.WithCancel(ctx)
+	cancel()
```
Comment (bassosimone) [inline internal/cmd/oonireport/oonireport_test.go:76]:
```suggestion
	cancel() // fail immediately
```

**Pull request merged**